protocols/util/factories/avro_factory.py

```diff
@@ -0,0 +1,70 @@
+import sys
+
+import factory.fuzzy
+from factory.base import FactoryMetaClass, BaseFactory, BaseMeta
+
+
+def mock_schema(protocol):
+    """
+    :type protocol: Protocol
+    """
+    expected_schema = protocol.schema
+    basics = ['null', 'boolean', 'string', 'bytes', 'int', 'long', 'float', 'double', 'enum']
+    mock_result = {}
+    for field in expected_schema.fields:
+        schema_type = field.type.type
+        name = field.name
+        if schema_type in basics:
+            basic_types(field, mock_result, name, schema_type)
+        elif schema_type == 'union':
+            basic_types(field.type.schemas[1], mock_result, name, field.type.schemas[1].type)
```

Comment:
I guess this means we have to always set `union {null, string}` and never union {string, null}. Couldn't we just get the first in the list not being null?